ote/src/clj/ote/transit_changes/detection.clj

```diff
@@ -1030,15 +1033,19 @@
   fetches and analyzes packages for the service and produces a collection of structures, each of which describes
   if a route has traffic or changes/no-traffic/ending-traffic, during a time period defined in the analysis logic.
   Output: ::detected-route-changes-for-services-coll"
-  (let [type (db-route-detection-type db service-id)
-        ;; Generate "key" for all routes. By default it will be a vector ["<route-short-name>" "<route-long-name" "trip-headsign"]
+  (let [;; Generate "key" for all routes. By default it will be a vector ["<route-short-name>" "<route-long-name" "trip-headsign"]
         service-routes (sort-by :route-hash-id (service-routes-with-date-range db {:service-id service-id}))
         all-routes (map-by-route-key service-routes type)
         all-route-keys (set (keys all-routes))
-        route-hashes (service-route-hashes-for-date-range db route-query-params)
+        route-hashes (sort-by :date
```

Comment:
Instead of`apply concat mapv fn` would this work `mapcat fn` ?

Reply:
Tried. Didn't work